lib/KBox/src/pages/NavigationPage.cpp

```diff
@@ -0,0 +1,305 @@
+        //DEBUG("Setting buttonPressed to true");
+    }
+    
+    if (be.clickType == ButtonEventTypeReleased && this->buttonPressed == true) {
+        if (this->buttonPressedTimer < 2000) {
+            // short click - return false to force the MFD to skip to the next page
+            //DEBUG("short click");
+            return false;
+        }
+        else if (this->buttonPressedTimer < 5000) { //medium click, if navMode is on, activate dodge mode
+            //DEBUG("medium click");
+            if (apMode == true){
+                if (apDodgeMode == true){
+                    apDodgeMode = false;
+                    if (apHeadingMode == true){
+                        apModeString = "APMode: Heading";
+                    } else if (apWaypointMode == true){
+                        apModeString = "APMode: Waypoint";
+                    }
+                    apModeDisplay->setText(apModeString);
+                    apModeDisplay->setColor(ColorGreen);
+                } else {
+                    apDodgeMode = true;
+                    apModeString = "APMode: Dodge     ";//extra spaces needed to overwrite old string if longer
+                    apModeDisplay->setText(apModeString);
+                    apModeDisplay->setColor(ColorRed);
+                }
+                return true;
+            } else {
+                DEBUG("apMode not active so no effect");
+            }
+        } else if (this->buttonPressedTimer > 5000) {  // Long click - start and stop the navMode here
+            DEBUG("long click");
+            if (apMode == false){
+                apMode = true;
+                apHeadingMode = true;
+                apModeString = "APMode: Heading   ";//extra spaces needed to overwrite old string if longer
+                apModeDisplay->setText(apModeString);
+                apModeDisplay->setColor(ColorGreen);
+            } else {
+                apMode = false;
+                apDodgeMode = false;
+                apModeString = "APMode: Off           ";
+                apModeDisplay->setText(apModeString);
+                apModeDisplay->setColor(ColorBlue);
+            }
+            return true;
+        }
+    }
+    //DEBUG("APmode is: - %d", apMode);
+    //DEBUG("APHeadingmode is: - %d", apHeadingMode);
+    //DEBUG("APDodgemode is: - %d", apDodgeMode);
+    return true;
+}
+
+bool NavigationPage::processEvent(const EncoderEvent &ee) {
+        encoderClicks = ee.rotation;
+        navEncoderClicks = navEncoderClicks + encoderClicks;
+        encoderClicks = 0;
+    
+        //DEBUG("encoderClicks - %d", encoderClicks);
+        //DEBUG("navEncoderClicks - %d", navEncoderClicks);
+    return true;
+}
+
+
+String NavigationPage::formatMeasurement(float measure, const char *unit) {
+  // extra spaces at the end needed to clear previous value completely
+  // (we use a non-fixed width font)
+  char s[10];
+  //snprintf(s, sizeof(s), "%.1f %s  ", measure, unit);
```

Comment:
removed, I now realize I did change it to get rid of the decimals on course headings